centreon/src/Core/Service/Infrastructure/Repository/DbWriteServiceActionLogRepository.php

```diff
@@ -0,0 +1,266 @@
+use Core\Service\Domain\Model\NotificationType;
+use Core\Service\Domain\Model\Service;
+use Core\Service\Infrastructure\Model\NotificationTypeConverter;
+
+class DbWriteServiceActionLogRepository extends AbstractRepositoryRDB implements WriteServiceRepositoryInterface
+{
+    use LoggerTrait;
+    public const SERVICE_OBJECT_TYPE = 'service';
+
+    /**
+     * @param WriteServiceRepositoryInterface $writeServiceRepository
+     * @param ContactInterface $contact
+     * @param ReadServiceRepositoryInterface $readServiceRepository
+     * @param WriteActionLogRepositoryInterface $writeActionLogRepository
+     * @param DatabaseConnection $db
+     */
+    public function __construct(
+        private readonly WriteServiceRepositoryInterface $writeServiceRepository,
+        private readonly ContactInterface $contact,
+        private readonly ReadServiceRepositoryInterface $readServiceRepository,
+        private readonly WriteActionLogRepositoryInterface $writeActionLogRepository,
+        DatabaseConnection $db
+    ) {
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete(int $serviceId): void
+    {
+        try {
+            $service = $this->readServiceRepository->findById($serviceId);
+            if ($service === null) {
+                throw new RepositoryException(sprintf('Cannot find service to delete (ID: %d).', $serviceId));
+            }
+            
+            $this->writeServiceRepository->delete($serviceId);
+
+            $actionLog = new ActionLog(
+                self::SERVICE_OBJECT_TYPE,
+                $serviceId,
+                $service->getName(),
+                ActionLog::ACTION_TYPE_DELETE,
+                $this->contact->getId()
+            );
+            $this->writeActionLogRepository->addAction($actionLog);
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+
+            throw $ex;
+        }
+    } 
+
+    /**
+     * @inheritDoc
+     */
+    public function deleteByIds(int ...$serviceIds): void
+    {
+        $failedDeletions = [];
+        foreach ($serviceIds as $serviceId) {
+            try {
+                $service = $this->readServiceRepository->findById($serviceId);
+                if ($service === null) {
+                    throw new RepositoryException('Cannot find service to delete');
+                }
+                
+                $this->writeServiceRepository->delete($serviceId);
+
+                $actionLog = new ActionLog(
+                    self::SERVICE_OBJECT_TYPE,
+                    $serviceId,
+                    $service->getName(),
+                    ActionLog::ACTION_TYPE_DELETE,
+                    $this->contact->getId()
+                );
+                $this->writeActionLogRepository->addAction($actionLog);
+            } catch (\Throwable $ex) {
+                $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
```

Comment:
you can specify business information with $serviceIds in the context of log to have more informations from the log

Reply:
Done

Reply:
Yes it's good, thanks for your change ;)